controller/Dockerfile

```diff
@@ -0,0 +1,7 @@
+FROM ubuntu:bionic
```

Comment:
This should do the same thing as https://github.com/RespiraWorks/VentilatorSoftware/blob/master/.travis.yml, right?
Two requests:
1) Please mention that file in a comment here
2) Looks like some of the commands are different: e.g. "platformio update" is present there but absent here; pre-commit is pinned here but not there. Is there a reason for the differences?

Reply:
👍 re: platformio stuff.

pre-commit stuff should be done in a different circleCI job

Reply:
>     1. Please mention that file in a comment here
>     2. Looks like some of the commands are different: e.g. "platformio update" is present there but absent here; pre-commit is pinned here but not there. Is there a reason for the differences?

I'll add the necessary comments. pre-commit is pinned here to address an issue with the latest version not being compatible (though apparently still install-able) with older Python interpreters.

Reply:
> 👍 re: platformio stuff.
> 
> pre-commit stuff should be done in a different circleCI job

Ok, I'll look into how to do that (probably Monday.)

Reply:
@hhff I created https://github.com/RespiraWorks/VentilatorSoftware/issues/209 to address the pre-commit ask.